### DISCUSSION TITLE
Running an error in maya2020-------TOOLS\utils\UI.py line 5: cannot import name GeneralWindow #

### DISCUSSION THREAD

**Discussion by zhai23** [GenEugene/GETools#39] [Report a problem] [answered]
Running an error in maya2020
# 错误: ImportError: file C:/Users/17893/Documents/maya/2020/zh_CN/scripts/TOOLS\utils\UI.py line 5: cannot import name GeneralWindow #

**Comment by GenEugene** [comment]
Hi. Did you try to run in on Maya 2023 or higher? I didn't test in on previous versions before 2023.

I think, I should do it later, but not sure how many changes autodesk did with api

**Comment by GenEugene** [comment]
And did you download all files from repository and they placed inside proper folders?

In my plugin you can't use scripts separate from other utils and modules, only all together with strict folder hierarchy. 

Just curious 😅

Looks like I still need to download 2020 maya version

**Comment by GenEugene** [comment]
@zhai23 check new 0.0.8 version. Probably it can fix your problem. I did some namespace fixes for package isolation to avoid any intersections with other scripts which could cause code conflicts.

Still not tested on your version, plan to do it later

**Comment by GenEugene** [comment]
So, i tried to run getools on different maya versions, 2020-2022-2023-2024
Only 2020 has an issue. I think it's because of Maya's python migration when they moved from 2020 to 2022. I remember this time, when a lot of people have troubles with theirs tools. 

So, I'm not sure, how to fix it fast, need investigate each module and find some logic what can't run on old maya.

**Comment by GenEugene** [comment, marked as answer]
So, finally I found a source of issue. Looks like I fixed it and it works well on 2020 version. Problem was in some cross dependencies. New maya versions have no critical problems with it, but old maya versions have something like block for potential circular cross imports. 

Fix will be available on v0.0.9

**Reply by zhai23** [reply]
This is great. I'm sorry that I haven't brought my computer with me recently. I forgot about it. I'll try it right away tonighti✨

**Reply by zhai23** [reply]
Perfect✨
